core/string/ustring.cpp

```diff
@@ -2575,6 +2575,72 @@ int64_t String::to_int(const wchar_t *p_str, int p_len) {
 	return integer * sign;
 }
 
+uint64_t String::to_uint64(bool p_clamp) const {
+	int str_length = length();
+	if (str_length == 0) {
+		return 0;
+	}
+
+	int start = 0;
+	// Skip leading whitespace
+	while (start < str_length &&
+			(is_whitespace(operator[](start)) || operator[](start) == '+')) {
+		start++;
+	}
+
+	// Check for negative sign
+	if (start < str_length && operator[](start) == '-') {
```

Comment:
Same here, is `++-2` an acceptable string to parse?

Reply:
with clamp it would eval to 0 without it would error as unsupported. Again thought about keeping the parser simple with less conditionals to check.